crates/tracing-instrumentation/src/lib.rs

```diff
@@ -96,28 +99,26 @@ where
     let mut tracer_provider_builder = opentelemetry_sdk::trace::TracerProvider::builder()
         .with_config(opentelemetry_sdk::trace::Config::default().with_resource(resource));
 
-    if let Some(endpoint) = &common_opts.tracing.tracing_endpoint {
-        let header_map =
-            HeaderMap::from_iter(HashMap::from(common_opts.tracing.tracing_headers.clone()));
+    let header_map = HeaderMap::from_iter(HashMap::from(opts.headers.clone()));
 
-        let exporter = SpanExporterBuilder::from(
-            opentelemetry_otlp::new_exporter()
-                .tonic()
-                .with_endpoint(endpoint)
-                .with_metadata(MetadataMap::from_headers(header_map)),
-        )
-        .build_span_exporter()?;
-        let exporter = ResourceModifyingSpanExporter::new(exporter);
-        tracer_provider_builder = tracer_provider_builder.with_span_processor(
-            BatchSpanProcessor::builder(exporter, opentelemetry_sdk::runtime::Tokio).build(),
-        );
-    }
+    let exporter = SpanExporterBuilder::from(
+        opentelemetry_otlp::new_exporter()
+            .tonic()
+            .with_endpoint(&opts.endpoint)
+            .with_metadata(MetadataMap::from_headers(header_map)),
+    )
+    .build_span_exporter()?;
+    let exporter = ResourceModifyingSpanExporter::new(exporter);
+
+    tracer_provider_builder = tracer_provider_builder.with_span_processor(
+        BatchSpanProcessor::builder(exporter, opentelemetry_sdk::runtime::Tokio).build(),
+    );
 
-    if let Some(path) = &common_opts.tracing.tracing_json_path {
+    if let Some(path) = &opts.json_path {
         let exporter = JaegerJsonExporter::new(
             path.into(),
             "trace".to_string(),
-            service_name,
+            SERVICE_NAME.into(),
```

Comment:
Did we come to a conclusion whether there is a problem with two exporters writing to the same output file? If we didn't default to the tracing runtime settings, then people wouldn't run into this problem (assuming it is a problem) so easily.